libdnf/transaction/Swdb.cpp

```diff
@@ -71,7 +91,6 @@ Swdb::Swdb(const std::string &path)
                 conn->exec("BEGIN; UPDATE config SET value='test' WHERE key='test'; ROLLBACK;");
             } catch (SQLite3::Error & ex) {
                 // root must have the database writable -> log and re-throw the exception
-                auto logger(libdnf::Log::getLogger());
```

Comment:
Why did you replace it? I believe that it will cause additional problem, because the logger will be loaded before it is fully initialized.

Reply:
It won't. The getter doesn't do any loading, it simply returns a static variable. By moving it to the beginning of the constructor doesn't really change anything. The getter would simply be called a bit earlier in the constructor, but I think it's going to be optimized out by the compiler anyway.

Reply:
The difference is really minor. It means that the logger is always imported even when it is not required. It means that the code will be slightly slower. The change has no benefit.

Reply:
What do you mean by "imported"? There is no difference at all. The getter only returns a pointer and that is stored in a local variable. There is absolutely no semantic difference and as I've said already, the getter itself is most likely optimized out by the compiler.

In case the getter wouldn't be optimized out, the performance impact is totally negligible. The benefit is saving multiple lines with the same content in the source.

Unless I'm missing something, it's really quite pointless arguing about this.